catalyst/core/utils/trace.py

```diff
@@ -0,0 +1,104 @@
+from typing import TYPE_CHECKING, Dict, Union  # isort:skip
+from pathlib import Path
+
+from catalyst.dl import Experiment
```

Comment:
Here I have an error `ImportError: cannot import name 'Experiment'` because of circular dependent imports. How is better to fix it?

Reply:
could you place this file to `catalyst/dl/utils`, there should not be any circular dependencies then

Reply:
It was in dl/utils, then I moved it to core/utils, so it won't help, already checked

Reply:
nevertheless by design of this function, it should be into `dl/utils`
simple logic:  we have `core`,  `dl` uses `core`, this function improves `dl` module and uses `dl` primitives (`dl` uses `core` one, but `core` could not use `dl`), so it should be under `dl` module :)

Reply:
Done, but still have circular dependent imports :)